src/App.js

```diff
@@ -7,17 +7,28 @@ import MyReservations from 'components/MyReservations';
 import CarList from 'components/home/CarList';
 import Login from 'components/session/Login';
 import SignUp from 'components/session/SignUp';
+import ReservationForm from 'components/ReservationForm';
+import { useDispatch } from 'react-redux';
+import { getCars } from 'redux/cars/carsSlice';
+import { useEffect } from 'react';
 
-const App = () => (
-  <>
-    <NavigationPanel />
-    <Routes>
-      <Route path="/" element={<Home />} />
-      <Route path="cars" element={<CarList />} />
-      <Route path="reservations" element={<MyReservations />} />
-      <Route path="login" element={<Login />} />
-      <Route path="signup" element={<SignUp />} />
-    </Routes>
-  </>
-);
+const App = () => {
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(getCars('CARS'));
+  }, [dispatch]);
```

Comment:
While you made no changes to the useEffect hook, however, you made changes to the  "CarList" component not created by you, and without prior notification. Furthermore, these changes were not neccessary except otherwise. Again, it's a good practice to keep the App component as light as possible. The useEffect hook will always trigger when the component mounts irrespective of where it is called. The getCars function was called on the **CarList**  component. Please restore the changes you made to the CarList component.

Reply:
hi @ernestmusong  

sorry for not notifying you but what I need is car data and fir a person directly visits the form page it won't have the data since we don't have useEffect hook on Top level that's why I change it to top level 

looking for suggestion on this 

😁😁🙌🙌

Reply:
Dear @PowerLevel9000 I understand your concern, however note that the cars data is made available to any component in the application through the **useSelector()** hook since it has direct access to the state. Remember that the cars data was already made available to the application the first time the component mounted.